include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -471,6 +471,46 @@ def NVMMASharedEncodingAttr :
   let hasCustomAssemblyFormat = 1;
 }
 
+def SwizzledBlocksSharedEncodingAttr :
+  TritonGPU_Attr<"SwizzledBlocksSharedEncoding", "swizzled_blocks_shared_encoding", [SharedEncodingTrait, LayoutEncodingTrait]> {
+  let mnemonic = "swizzled_blocks_shared";
+
+  let description = [{
+Shared encoding similar to SwizzledSharedEncodingAttr,
+but with additional swizzling pattern change in blocks of maxPhase * perPhase rows:
+
+  #shared<{vec=1, perPhase=1, maxPhase=4, order=[1, 0]}>
```

Comment:
It's unclear how the block # is calculated in the below. Can you add text to elaborate for this example?